src/lib/barcode_matching.rs

```diff
@@ -80,17 +80,20 @@ impl BarcodeMatcher {
     }
 
     /// Counts the number of bases that differ between two byte arrays.
-    fn count_mismatches(observed_bases: &[u8], expected_bases: &[u8]) -> u8 {
+    fn count_mismatches(observed_bases: &BitEnc, expected_bases: &BitEnc) -> u8 {
         assert_eq!(
-            observed_bases.len(),
-            expected_bases.len(),
+            observed_bases.nr_symbols(),
+            expected_bases.nr_symbols(),
             "observed_bases: {}, expected_bases: {}",
-            observed_bases.len(),
-            expected_bases.len()
+            observed_bases.nr_symbols(),
+            expected_bases.nr_symbols()
         );
         let mut count: usize = 0;
-        for (&expected_base, &observed_base) in expected_bases.iter().zip(observed_bases.iter()) {
-            if !byte_is_nocall(expected_base) && expected_base != observed_base {
+        for i in 0..observed_bases.nr_symbols() {
+            let observed_base = observed_bases.get(i).unwrap();
+            let expected_base = expected_bases.get(i).unwrap();
+            // Note: this allows IUPAC fuzzy matching with IUPAC bases in the expected barcodes, but not in the observed barcodes.
+            if expected_base & observed_base != observed_base {
```

Comment:
Technically I think this will allow IUPAC in the observed ... so long as it's at least as specific at the expected, right?  For example, if both are `N` then it will pass.  But if expected is `R` and observed is `N` it will fail?